scilpy/image/operations.py

```diff
@@ -112,396 +113,470 @@ def _validate_float(x):
-    ref_array = input_list[0]
-
-    if len(input_list) == 1 and not ref_array.ndim > 3:
-        logging.error('This operation with only one operand requires 4D data.')
-        raise ValueError
+    _validate_imgs(*input_list, ref_img)
 
-    in_data = np.squeeze(np.rollaxis(np.array(input_list), 0,
-                                     input_list[0].ndim+1))
+    if len(input_list[0].header.get_data_shape()) > 3:
+        if not len(input_list) == 1:
+            raise ValueError(
+                'This operation with 4D data only support one operand.')
+    else:
+        if len(input_list) == 1:
+            raise ValueError(
+                'This operation with only one operand requires 4D data.')
 
-    return np.average(in_data, axis=-1)
+    if len(input_list[0].header.get_data_shape()) > 3:
+        return np.average(input_list[0].get_fdata(dtype=np.float64), axis=-1)
+    else:
+        return addition(input_list, ref_img) / len(input_list)
 
 
-def std(input_list):
+def std(input_list, ref_img):
     """
     std: IMGs
         Compute the standard deviation average of multiple images.
         If a single 4D image is provided, compute the STD along the last
         dimension.
     """
     _validate_length(input_list, 1, at_least=True)
-    _validate_arrays(*input_list)
-    ref_array = input_list[0]
-
-    if len(input_list) == 1 and not ref_array.ndim > 3:
-        logging.error('This operation with only one operand requires 4D data.')
-        raise ValueError
+    _validate_imgs(*input_list, ref_img)
 
-    in_data = np.squeeze(np.rollaxis(np.array(input_list), 0,
-                                     input_list[0].ndim+1))
+    if len(input_list[0].header.get_data_shape()) > 3:
+        if not len(input_list) == 1:
+            raise ValueError(
+                'This operation with 4D data only support one operand.')
+    else:
+        if len(input_list) == 1:
+            raise ValueError('This operation with only one operand requires 4D data.')
 
-    return np.std(in_data, axis=-1)
+    if len(input_list[0].header.get_data_shape()) > 3:
+        return np.std(input_list[0].get_fdata(dtype=np.float64), axis=-1)
+    else:
+        mean_data=mean(input_list, ref_img)
+        output_data=np.zeros(input_list[0].header.get_data_shape())
+        for img in input_list:
+            if isinstance(img, nib.Nifti1Image):
+                data=img.get_fdata(dtype=np.float64)
+                output_data += (data - mean_data) ** 2
+                img.uncache()
+            else:
+                output_data += (img - mean_data) ** 2
+        return np.sqrt(output_data / len(input_list))
 
 
-def union(input_list):
+def union(input_list, ref_img):
     """
     union: IMGs
         Operation on binary image to keep voxels, that are non-zero, in at
         least one file.
     """
-    output_data = addition(input_list)
-    output_data[output_data != 0] = 1
+    output_data=addition(input_list, ref_img)
```

Comment:
You had a formatting bug I guess, you are missing spaces around operators for no reason, check everywhere please.